app/javascript/components/WorkflowsPage/index.tsx

```diff
@@ -0,0 +1,116 @@
+  navigation?: React.ReactNode;
+  children: React.ReactNode;
+  preview?: React.ReactNode;
+};
+
+export const Layout = ({ title, actions, navigation, children, preview }: LayoutProps) => (
+  <>
+    <PageHeader className="sticky-top" title={title} actions={actions}>
+      {navigation ?? null}
+    </PageHeader>
+    {preview ? (
+      <div className="fixed-aside flex-1 lg:grid lg:grid-cols-[1fr_30vw]">
+        <div>{children}</div>
+        <aside className="hidden lg:block" aria-label="Preview">
+          {preview}
+        </aside>
+      </div>
+    ) : (
+      <div>{children}</div>
+    )}
+  </>
+);
+
+type PublishButtonProps = {
+  isPublished: boolean;
+  wasPublishedPreviously: boolean;
+  isDisabled: boolean;
+  sendToPastCustomers: {
+    enabled: boolean;
+    toggle: (value: boolean) => void;
+    label: string;
+  } | null;
+  onClick: (saveActionName: SaveActionName) => void;
+};
+
+export const PublishButton = ({
+  isPublished,
+  wasPublishedPreviously,
+  isDisabled,
+  sendToPastCustomers,
+  onClick,
+}: PublishButtonProps) => {
+  const [popoverOpen, setPopoverOpen] = React.useState(false);
+
+  return isPublished ? (
+    <Button onClick={() => onClick("save_and_unpublish")} disabled={isDisabled}>
+      Unpublish
+    </Button>
+  ) : wasPublishedPreviously || sendToPastCustomers === null ? (
+    <Button color="accent" onClick={() => onClick("save_and_publish")} disabled={isDisabled}>
+      Publish
+    </Button>
+  ) : (
+    <Popover
+      disabled={isDisabled}
+      trigger={
+        <div className="button" color="accent">
+          Publish
+          <Icon name="outline-cheveron-down" />
+        </div>
+      }
+      open={popoverOpen}
+      onToggle={setPopoverOpen}
+    >
+      <fieldset>
+        <Button color="accent" onClick={() => onClick("save_and_publish")} disabled={isDisabled}>
+          Publish now
+        </Button>
+        <Toggle value={sendToPastCustomers.enabled} onChange={sendToPastCustomers.toggle}>
+          {sendToPastCustomers.label}
+        </Toggle>
+      </fieldset>
+    </Popover>
+  );
+};
+
+export const sendToPastCustomersCheckboxLabel = (workflowTrigger: WorkflowTrigger) =>
+  workflowTrigger === "new_subscriber"
+    ? "Also send to past email subscribers"
+    : workflowTrigger === "member_cancels"
+      ? "Also send to past members who canceled"
+      : workflowTrigger === "new_affiliate"
+        ? "Also send to past affiliates"
+        : "Also send to past customers";
+
+export const EditPageNavigation = (props: { workflowExternalId: string }) => {
+  const page = usePage();
+  const currentUrl = page.url;
+
+  return (
+    <Tabs>
+      <Tab isSelected={currentUrl.includes(`/workflows/${props.workflowExternalId}/edit`)} asChild>
+        <Link href={Routes.edit_workflow_url(props.workflowExternalId)}>Details</Link>
+      </Tab>
+      <Tab isSelected={currentUrl.includes(`/workflows/${props.workflowExternalId}/emails`)} asChild>
```

Comment:
Since this is now an actual backend route, we should use the appropriate url helpers instead of hard coded paths.